File: arcane-sdk-ts/src/resources/datasources.ts
import type { ArcaneClient } from '../client';
import type { DatasourceListItem } from '../types';

export class DatasourcesResource {
  constructor(private readonly client: ArcaneClient) {}

  async list(): Promise<DatasourceListItem[]> {
    return this.client.get<DatasourceListItem[]>('/api/public/datasources');
  }
}

File: arcane-sdk-py/src/arcane_sdk/resources/datasources.py
"""Datasources resource."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from arcane_sdk.client import ArcaneClient


class DatasourcesResource:
    def __init__(self, client: ArcaneClient) -> None:
        self._client = client

    def list(self) -> list[dict[str, Any]]:
        """List all datasources."""
        return self._client._get("/api/public/datasources")
